content/06.model-complexity.md

```diff
@@ -1,34 +1,34 @@
-Recent studies suggest that there are limitations to decision tree-based ensemble methods when applied to rare disease datasets with substantial class imbalance and label-noise [@doi:10.1007/s11634-019-00354-x; @pmid:30815073]. 
-This has led to the adoption of cascade learning, a variant of ensemble learning, where multiple methods leveraging distinct underlying assumptions are used in tandem; and augmented with algorithms like AdaBoost (boosting) to capture stable patterns existing in silver standard data [@doi:10.1109/cvpr.2001.990537; @doi:10.1007/978-3-540-75175-5_16; @doi:10.1109/icpr.2004.1334680]. 
+It should be noted here that resampling approaches are most helpful in constructing confidence intervals for algorithms that generate the same outcome everytime they are run (i.e. deterministic models).
+For non-deterministic models like decision trees that choose features at random for selecting a path to the outcome, resampling approaches can be helpful in determining robustness or stability of the model. 
+
+Recent studies suggest that there are limitations to decision tree-based ensemble methods when applied to rare disease datasets, leading to adoption of cascade learning, a variant of ensemble learning.[@doi:10.1007/s11634-019-00354-x; @pmid:30815073]
+In cascade learning, multiple methods leveraging distinct underlying assumptions are used in tandem to capture stable patterns existing in the dataset [@doi:10.1109/cvpr.2001.990537; @doi:10.1007/978-3-540-75175-5_16; @doi:10.1109/icpr.2004.1334680]. 
 For example, a cascade learning approach for identifying rare disease patients from electronic health record data utilized independent steps for feature extraction (word2vec [@arxiv:1301.3781]), preliminary prediction with ensembled decision trees, and prediction refinement using data similarity metrics [@pmid:30815073]. 
-Combining these three methods resulted in better performance than other methods when implemented on the silver standard dataset in isolation.
-Other approaches like inverse class weighting, oversampling (with methods like Synthetic Minority Oversampling Technique, abbreviated SMOTE), or cascade learning approaches that adequately represent the rarer classes are also beneficial in increasing stability of predictions while accommodating for class imbalance.
+Combining these three methods resulted in better performance than other methods when implemented on a silver standard dataset in isolation.
+Other approaches like inverse class weighting, oversampling (with methods like Synthetic Minority Oversampling Technique, abbreviated SMOTE), or methods that adequately represent the rarer classes are also beneficial in increasing stability of predictions while accommodating for class imbalance.
```

Comment:
As a new-to-this person, I don't think I'd know what inverse class weighting or oversampling mean. Should we bother mentioning these if we don't explain them, or if we don't have any examples of them being used in rare disease?

Or maybe just reordering the sentence to make it gentler: 

```suggestion
Other approaches that better represent rare classes like inverse class weighting, oversampling [doi:10.1613/jair.953], or cascade learning may also improve ML models that use rare disease data. 
```

"methods that adequately represent the rarer classes " I put cascade learning back in because this describes all of these methods, not just cascade learning.

Reply:
I also removed SMOTE - we can just cite it since we don't talk about it elsewhere AFAIK

Reply:
Agree with Robert's suggestions